### PR TITLE
fix(#2562): Prevent transaction deduplication for imported transactions

### PR DESCRIPTION
**Pull request by pmoon00**
Resolves https://github.com/actualbudget/actual/issues/2562

Based on https://github.com/actualbudget/actual/pull/2618 , with comments by @MatissJanis and others applied

Resolves issues that caused a rollback [described here](https://github.com/actualbudget/actual/issues/2877)

This resolves the goCardless fuzzy search predicament by having two types of fuzzy search behaviours. The default fuzzy search doesn't allow for transactions with different import IDs from being matched. The old fuzzy logic remains to handle goCardless's quirk.

### CODE REVIEW

**Comment by pmoon00** [comment]
@MatissJanis @strazto @matt-fidd I've added changes to have both fuzzy searches. I've defaulted the fuzzy search to default to the new logic, while the goCardless syncs will use the old fuzzy search logic.

Happy to get some feedback on this.

**Comment by matt-fidd** [comment]
@pmoon00 is this still WIP? It looks like this doesn't build as-is.

**Comment by pmoon00** [comment]
> @pmoon00 is this still WIP? It looks like this doesn't build as-is.

It was working but then the latest rebase from master looks to have broken it. I will look to fix it very soon.

**Comment by pmoon00** [comment]
> @pmoon00 is this still WIP? It looks like this doesn't build as-is.

@matt-fidd I've sorted out the issue, ready for review :)

**Review by matt-fidd — APPROVED** [review]
I'm happy with this, my only concern is introducing inconsistent behaviour but I believe it's worth it. 

@MatissJanis could you weigh in on this as an additional maintainer? You were involved in the original change is all.

File: packages/loot-core/src/server/accounts/sync.ts [hunk 1]
```diff
@@ -620,6 +646,10 @@ export async function syncAccount(
   );
 
   const acctRow = await db.select('accounts', id);
+  // If syncing an account from Go Cardless it must not use strictIdChecking. This allows
+  // the fuzzy search to match transactions where the import IDs are different. It is a known quirk
+  // that Go Cardless can give two different transaction IDs even though it's the same transaction.
+  const useStrictIdChecking = acctRow.account_sync_source !== 'goCardless';
```
Comment (MatissJanis) [inline packages/loot-core/src/server/accounts/sync.ts:652]: **❓ question:** ‏shouldn't this also exclude SimpleFIN?

Comment (pmoon00) [reply packages/loot-core/src/server/accounts/sync.ts:652]: I'm not sure. I was fixing this based on the problems described in [this issue](https://github.com/actualbudget/actual/issues/2877). If Simple FIN is also affected in the same way as GoCardless I can add it in for that too. Please just let me know 🙏🏼

Comment (MatissJanis) [reply packages/loot-core/src/server/accounts/sync.ts:652]: Yea, I think the same thing should be applicable to SimpleFIN (and all future bank-sync providers). So I think the condition can be changed to 
```
const useStrictIdChecking = !acctRow.account_sync_source;
```

**Review by MatissJanis — APPROVED** [review]

**Review by matt-fidd — APPROVED** [review]

**Comment by psybers** [comment]
This PR caused a bug with importing CSV.  See #3227 for details.

I confirmed that bug.  I then rolled my master back one commit and the bug went away.

**Pull request merged**